vllm/model_executor/models/llama_eagle3.py

```diff
@@ -215,6 +215,9 @@ def compute_logits(
         logits = self.logits_processor(self.lm_head, hidden_states,
                                        sampling_metadata)
         if self.draft_id_to_target_id is None:
+            assert logits.shape[1] == self.config.vocab_size, \
+                "Expected logits to have shape " \
+                f"(*, {self.config.vocab_size}), but got {logits.shape}"
```

Comment:
A dumb question: Now that we don't set `self.draft_id_to_target_id = None`, how can this branch be taken?

Reply:
@benchislett ^ Could you please answer this question?